src/core/bitop.d

```diff
@@ -142,6 +156,74 @@ unittest
     assert(test_ctfe == 63);
 }
 
+private alias softBsf(N) = softScan!(N, true);
+private alias softBsr(N) = softScan!(N, false);
+
+private int softScan(N, bool forward)(N v) pure
+    if(is(N == uint) || is(N == ulong))
+{
+    enum mask(ulong lo) = forward? cast(N) lo : cast(N)~lo;
+    enum inc(int up) = forward? up : -up;
+
+    if (!v)
+        return -1;
+
+    N x = void;
+    int ret = void;
+    static if (is(N == ulong))
+    {
+        x = v & mask!0x0000_0000_FFFF_FFFFL;
+        if (x)
+        {
+            v = x;
+            ret = forward? 0 : 63;
+        }
+        else
+            ret = forward? 32 : 31;
+
+        x = v & mask!0x0000_FFFF_0000_FFFFL;
+        if (x)
+            v = x;
+        else
+            ret += inc!16;
+    }
+    else
```

Comment:
else what values of N are acceptable? Clearly `ubyte` won't work, but there needs to be a test for what the code is designed to work with.